File: ClientApp/src/app/Models/message.model.ts
import { ChatUserModel } from "./chat-user.model";
import { MessageReplyModel } from "./mesage-reply.model";

export class MessageModel{
  messageId:number;
  body:string;
  activeStatusId:number;
  createdDate :string;
  updatedDate: string;
  user:ChatUserModel;
  messageReplys:MessageReplyModel[];
}

File: Models/MessageModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace HP_Messaging.Models
{
    public class MessageModel
    {
        public int MessageId { get; set; }
        [Required]
        public string Body { get; set; }
        [Required]
        public int ActiveStatusId { get; set; }
        public string CreatedDate { get; set; }
        public string UpdatedDate { get; set; }
        public virtual UserModel User { get; set; }
        public virtual List<MessageReplyModel> MessageReplys { get; set; }
    }
}
